File: algorithms/c/108.c
#include <stdio.h>
#include <stdlib.h>

struct TreeNode
{
    int val;
    struct TreeNode *left;
    struct TreeNode *right;
};

struct TreeNode* sortedArrayToBST(int* nums, int numsSize)
{
    if ( numsSize == 0 )
        return NULL;
    struct TreeNode *tree = (struct TreeNode *)malloc(sizeof(struct TreeNode));
    int m = numsSize/2;
    tree->val = nums[m];
    tree->left = sortedArrayToBST(nums, m);
    tree->right = sortedArrayToBST(nums+m+1, numsSize-m-1);
    return tree;
}

int main()
{
    return 0;
}

File: algorithms/python/108.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def sortedArrayToBST(self, nums):
        """
        :type nums: List[int]
        :rtype: TreeNode
        """
        def helper(nums):
            if len(nums) == 0: return None
            idx = len(nums) // 2
            node = TreeNode(nums[idx])
            node.left = helper(nums[:idx])
            node.right = helper(nums[idx+1:])
            return node
        return helper(nums)
